### ISSUE TITLE
root crontab gets populated with chef-client line after reboot

### ISSUE THREAD

**Issue by tsp0001** [chef-partners/azure-chef-extension#154]
After we successfully ran MS chef extension (RHEL7.2 VM), it populates root crontab with below line.Even if we remove that line from crontab, reboot brings it back.

```
[cwarfiel@cwarfiel cron]$ cat root
# Chef Name: azure_chef_extension
*/30 * * * * /bin/sleep 0; chef-client -c /etc/chef/client.rb -L /var/log/azure/Chef.Bootstrap.WindowsAzure.LinuxChefClient/1210.12.104.1000/chef-client.log --pid /etc/chef/azure-chef-client.pid --once
```

Is there a way to control that and keep it out of the crontab

**Comment by tsp0001** [comment]
Never mind, I fugured it out, thank you

**Comment by matkam** [comment]
@tsp0001 What did you figure out? I'm having an issue where chef-client is getting run every 30 minutes and found this in my crontab as well. I do not think this should be the default behavior.

**Comment by matkam** [comment]
Looks like running `/var/lib/waagent/Chef.Bootstrap.WindowsAzure.LinuxChefClient-1210.12.105.1000/disable.sh` removes the entry from root crontab. How do I keep it from getting added in the first place? I want it to "install" the extension and bootstrap chef, and not "enable" the extension.

**Comment by matkam** [comment]
I found the problem, looks like this commit took out the checks for "auto_update_false" setting before enabling the cron job: https://github.com/chef-partners/azure-chef-extension/commit/8aaedc7f94c771f3910c6c77dedae88b6414b96f

@NimishaS could I request having the check added back in? I think there should be two flags: one for auto updating azure-chef-extension, and a second one to enable the chef-client cron job.

**Comment by NimishaS** [comment]
@matkam , the purpose of `auto_update_false` was to stop the update of `chef-client` package but we found that `waagent` triggers the auto-update of the extension(which updates the chef-client version too if available) and  it's not configurable. 
If the value of this flag was false or not set during auto-update of the extension, then the old extension used to get uninstalled, but the new one was also not getting installed. This was causing issue. Hence this flag is removed.

Currently setting cronjob to run `chef-client` after 30 minutes is the designed behavior of the extension. We need to discuss with the Chef team for the flag suggested by you i.e. to enable the chef-client cron job.

**Comment by matkam** [comment]
For my purposes, it is fine if chef-client and the extension get auto updated. I just don't want chef-client to be run at all automatically. Thanks for the update @NimishaS. For now I have added the following in our base chef recipe as a workaround:
```
cron  'azure_chef_extension' do
  action :delete
end
```

**Comment by Aliasgar16** [comment]
@tsp0001 @matkam We have a implemented a new feature in `azure_chef_extension` where user can take decision to deploy or not to deploy `chef-client service` along with the extension, also user can configure the run interval value for `chef-client service`. Refer this [PR](https://github.com/chef-partners/azure-chef-extension/pull/161). Though this PR is already merged, latest extension version does not have the feature yet. We will soon publish the new version of the extension with this feature and will let you know. Thanks.

**Comment by Aliasgar16** [comment]
@tsp0001 @matkam We have fixed this in the latest extension version `1210.12.106.1000`.
